### ISSUE TITLE
Noise reduction fails when image contains a large uniform (i.e. overexposed) area

### ISSUE THREAD

**Issue by heckflosse** [Beep6581/RawTherapee#3469]
From here: https://discuss.pixls.us/t/noise-reduction-for-highlight-jpg/2301/11

The image contains a very large overexposed area which leads to noise reduction failing in preview and in all modes except `Auto multi-zones`.

I pushed a fix to [a new branch](https://github.com/Beep6581/RawTherapee/tree/denoise_uniform). I also checked before/after fix with a lot of files which don't contain a large uniform area. For these files I detected no differences visible by eyeball. Only differences using `amplified absolute Difference`.

Tests are welcome,

Ingo

**Comment by Beep6581** [comment]
Related: https://discuss.pixls.us/t/image-in-viewer-different-from-rawtherapee/2521

**Comment by heckflosse** [comment]
@Beep6581 

> Related: https://discuss.pixls.us/t/image-in-viewer-different-from-rawtherapee/2521

No, it's a different issue. The example from pixls.us is related to Denoise Luminance-Detail (Detail recovery)

**Comment by Hombre57** [comment]
I'll let you compare the result of PR #3470 against the dev branch here : https://filebin.net/1c97emtkw8yhtb7s

Overall, as you can see at the top left of this image, small saturated colors are slightly more smeared, but I don't know if it should prevent @heckflosse from committing the patch.

**Comment by heckflosse** [comment]
@Hombre57 I see what you mean. Maybe Jacques can chime in and test when he's back from vacancies @Desmis.
The patch is so old that some additional weeks won't matter at all :)

**Comment by heckflosse** [comment]
@Hombre57 I compared before/after of your file at 200%. To me it seems that with my changes the nr is a bit stronger without loosing relevant details. But that may be different for other files. We have to test further. Maybe @Beep6581 can also chime in.

**Comment by Desmis** [comment]
My grandsons have left since Sunday, and I am going to the Caribbean next Saturday...for 2 weeks :)

I looked at the patch proposed by Ingo and it seems to me that it provides a good solution to the problem.
Maybe there is another solution, but at the moment I do not see which.
Thanks to Ingo :)

**Comment by heckflosse** [comment]
@Desmis 

> and I am going to the Caribbean next Saturday...for 2 weeks :)

Lucky you!

Thanks for looking at my patch :+1: 

Ingo

**Comment by Desmis** [comment]
Ingo

Thank you :)

**Comment by Hombre57** [comment]
@Desmis @Beep6581 Any suggestion on what to do with this patch ? Is the result / trade-off acceptable for you ? It fix artifact on very specific images and make the denoising a little bit stronger on all images (AFAIK). It would be nice to have your though on your own test.

**Comment by Beep6581** [comment]
I have not tested this properly yet. Prefer to leave it for 5.2.

**Comment by heckflosse** [comment]
@Beep6581 Agreed

**Comment by Desmis** [comment]
Agreed

**Comment by agriggio** [comment]
Hi, what's the status of this? I think I found another "real-world" test case... :-)

**Comment by heckflosse** [comment]
@agriggio Alberto, the status didn't change, because it got not enough tests. Can you share your test case?

**Comment by Beep6581** [comment]
Still valid?

**Comment by heckflosse** [comment]
Imho still valid
